extensions/control-plane/store/sql/contract-definition-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractdefinition/SqlContractDefinitionStore.java

```diff
@@ -178,9 +208,69 @@ private long mapCount(ResultSet resultSet) throws SQLException {
         return resultSet.getLong(1);
     }
 
+    private Object fromPropertyValue(String value, String type) throws ClassNotFoundException {
+        var clazz = Class.forName(type);
+        if (clazz == String.class) {
+            return value;
+        }
+        return fromJson(value, clazz);
+    }
+
+    private SqlPropertyWrapper mapProperties(ResultSet resultSet) throws SQLException, ClassNotFoundException {
+        var name = resultSet.getString(statements.getContractDefinitionPropertyNameColumn());
+        var value = resultSet.getString(statements.getContractDefinitionPropertyValueColumn());
+        var type = resultSet.getString(statements.getContractDefinitionPropertyTypeColumn());
+        var isPrivate = resultSet.getBoolean(statements.getContractDefinitionPropertyIsPrivateColumn());
+        return new SqlPropertyWrapper(isPrivate, new AbstractMap.SimpleImmutableEntry<>(name, fromPropertyValue(value, type)));
+    }
+
     private ContractDefinition findById(Connection connection, String id) {
-        var sql = statements.getFindByTemplate();
-        return queryExecutor.single(connection, false, this::mapResultSet, sql, id);
+        return transactionContext.execute(() -> {
+            var query = QuerySpec.Builder.newInstance().filter(List.of(new Criterion("id", "=", id))).build();
+            var queryStatement = statements.createQuery(query);
+
+            var contractDefinitionStream = queryExecutor.query(connection, false,
+                            this::mapResultSet, queryStatement.getQueryAsString(), queryStatement.getParameters());
+            var allPropertiesStream = queryExecutor.query(connection, false,
+                            this::mapProperties, statements.getFindPropertyByIdTemplate(), id);
+
+            Map<Boolean, List<SqlPropertyWrapper>> groupedProperties =
+                        allPropertiesStream.collect(partitioningBy(SqlPropertyWrapper::isPrivate));
+            var contractDefinitionPrivateProperties = groupedProperties.get(true).stream().collect(
```

Comment:
this grouping is not needed, because properties can only be private. don't just copy-paste code!

Reply:
Grouping is removed.